### DISCUSSION TITLE
Use IntervalTree data Structuture in _find_genes.py

### DISCUSSION THREAD

**Discussion by rLannes** [colomemaria/epiScanpy#115] [Ideas]
I realize that for _find_genes.py you are using a scanning for loop to identify overlapping peaks genes.

An IntervalTree data structure will be way more efficient in terms of memory consumption and speed. 
We are speaking order of magnitude.

Open source library implementing this data structure exists.
Do you accept merge request?

**Comment by DaneseAnna** [comment]
Hi, 

Thank you for the feedback! 
Yes, we accept merge requests :D

**Reply by rLannes** [reply]
I worked on the gene_activity code as it is the one I am interested on.
Changing the data structure and your code a bit, I ended up a speed-up. At least 250x times.

I have some questions:
 
You need the intervaltree library than can be installed by pip, or using conda. 
Do I need to add it environment.yml?

Your current implementation for loading gtf is not working for all gtf files. For example, a gtf loaded from ensembl and using ensembl_havana annotation will fail using the current implementation.
Do you mind If I change that to?

Furthermore, you use transcript_name or gene_name as index this can lead to duplicate index name, IMO you should try to avoid that. A solution might be either keep index numerical or to make them unique. But that it is your call as It may make sense to keep thing as they are too.

Best,
Romain

**Reply by rLannes** [reply]
Lastly, 
do you have a standard dataset with a corresponding gtf file to test that I get the exact same results?
